specs/agents/tracing-distributed-tracing.md

```diff
@@ -191,6 +191,65 @@ Implementation note: The `traceparent` and `tracestate` specifications only allo
 This implementation guarantees backwards compatibility with our older agents and compatibility with current OpenTelemetry instrumentations.
 We will eventually remove the `elasticapmtraceparent` entirely as soon as we can safely assume that most users have already upgraded all their agents.
 
+### Baggage
+
+To propagate distributed context, we implement the [W3C Baggage](https://www.w3.org/TR/baggage/) specification.
+
+Agents MUST parse, validate, and attach the `baggage` header according [to the W3C specification](https://www.w3.org/TR/baggage/).
+
+In addition to the W3C specification, agents also SHOULD offer users 2 ways to use the values propagated via baggage:
+- Offer a Baggage API to manually manipulate and read baggage values - this is preferable the OpenTelemetry API.
+- Offer baggage related configuration to automatically store baggage values on events.
+
+#### Baggage API
+
+Agents SHOULD integrate with the existing OpenTelemetry API. Users should be able to use the OpenTelemetry API to interact with the baggage maintained by the agent. The primary way to interact with baggage SHOULD be the OpenTelemetry API.
+
+In case using the OpenTelemetry API isn't feasible in a given language, the agent SHOULD extend the proprietary agent API to interact with baggage.
+
+This API MUST offer the following functionalities:
+- Adding a new baggage item
+- Changing an existing baggage item
+- Removing an existing baggage item
+- Reading a specific baggage item
+- Reading all baggage items
+
+#### Baggage related configurations
+
+The following configurations enable users to automatically store baggage items on a given event without any code change. Baggage items with matching keys are stored in `otel.attributes`, except on errors.
+
+`baggage_to_attach_on_transactions` configuration
+
+A list of baggage keys which are automatically attached to the transaction. When a transaction is created, the agent iterates through all baggage items currently available and stores the ones with keys that match one of the items from the configured wildcard matcher list on the newly created transaction as an item in `otel.attributes`.
+
+|                |   |
+|----------------|---|
+| Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
+| Default        | `*`    |
```

Comment:
Changed the default compared to the original idea. We discussed with @AlexanderWert.

- Most requests won't have `baggage`, so most of the time it's not relevant
- We make it opt-out, and store everything by default
- Later we plan to integrate with synthetic - and with that, baggage keys used by synthetic should definitely be part of the default. Using `*` from the begging makes this easier: we always store everything by default, there is no need to adapt the default later for synthetic integration.

Reply:
After reading https://www.honeycomb.io/blog/ask-miss-o11y-opentelemetry-baggage (from https://github.com/elastic/apm/pull/820#discussion_r1285824331) I think we may need to come back to this and think about this again.

Reply:
We settled on making it opt-in since we prefix baggage, and the main concern was overwriting existing data in attributes. https://github.com/elastic/apm/pull/820#issuecomment-1671024912